### PR TITLE
Add __coll_repr__ to SyftSigningKey

### PR DESCRIPTION
**Pull request by jcardonnet**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by jcardonnet** [comment]
I added `_coll_repr_`  to `SyftSigningKey`  so now the list displays like this. 
However, it seems that there should be some sort of ID besides the value itself but I don't know which field is the right one.

image

for reference, `_repr_` just outputs `SyftSigningKey for: <verify_key>`

File: packages/syft/src/syft/node/credentials.py [hunk 1]
```diff
@@ -76,6 +76,11 @@ def from_string(key_str: str) -> SyftSigningKey:
     def __repr__(self) -> str:
         return f"<{SIGNING_KEY_FOR}: {self.verify}>"
 
+    def _coll_repr_(self) -> dict[str, str]:
+        return {
+            SIGNING_KEY_FOR: self.verify,
```
Comment (IonesioJunior) [inline packages/syft/src/syft/node/credentials.py:81]: Using `SIGNING_KEY_FOR` as a field name might be unclear in terms of what the value represents. I'd go for a more descriptive name, such as `KEY'S_PUBLIC_VALUE` or something similar in order to improve clarity.

Comment (jcardonnet) [reply packages/syft/src/syft/node/credentials.py:81]: are you suggesting to change the variable name of it's value?
`SIGNING_KEY_FOR = "SigningKey for"`

Comment (jcardonnet) [reply packages/syft/src/syft/node/credentials.py:81]: Agreed with @IonesioJunior to change the value to "Corresponding Public Key"

Comment (jcardonnet) [reply packages/syft/src/syft/node/credentials.py:81]: image

**Review by IonesioJunior — APPROVED** [review]

**Pull request merged**